manifests/certificate.pp

```diff
@@ -40,23 +43,35 @@
 ) {
   require ::acme::setup::common
 
-  $domain_dc = downcase($domain)
   $path = $acme::path
 
+  if $domain == undef {
+    # compatibility mode, parse name as list of domains, and use first as certificate resource name
+    $domains = split(downcase($name), ' ')
+    $cert_name = $domains[0]
+  } elsif $domain =~ String {
+    $domains = split($domain, ' ')
+    $cert_name = $name
+  } else {
+    $domains = $domain
+    $cert_name = $name
+  }
```

Comment:
I found a way to make this backwards-compatible after all. As long as the domain property of a certificate is not set, the name of the created resources will be the main certificate, as it was before.

Only if a certificate was created with both the resource name and the domain property set to a list of domains will this be a breaking change.